arroyo/processing/strategies/run_task.py

```diff
@@ -76,10 +73,21 @@ def poll(self) -> None:
 
             self.__queue.popleft()
 
-            message = Message(partition, offset, result, timestamp)
+            payload: BasePayload[TResult]
+            if isinstance(message.data, BrokerPayload):
+                payload = BrokerPayload(
+                    result,
+                    message.data.partition,
+                    message.data.offset,
+                    message.data.timestamp,
+                )
+            else:
+                payload = Payload(result, message.committable)
```

Comment:
This pattern risk to become concerning. 
I think we should try to avoid littering the code base with `isinstance` calls that run different logic for each type of payload. We will certainly have an incident sooner or later for forgetting one type when adding one payload type.

Options:
- there is a way with mypy to check exhaustiveness of an if block. https://hakibenita.com/python-mypy-exhaustive-checking. That is designed for enums, not sure we can do something with it for different implementations, but I would look into that. 
- have transformation functions that generate a `Message` from payload replacing some content (on the line of dataclass `replace` functions). At least this pattern will be in fewer places
- horrible (but likely better than this), add a `to_message` method in the payload class.

Reply:
Ok. Added the replace method to the `Payload` to support swapping out the data and preserving the rest